openshift/purge_AWS_resources.py

```diff
@@ -0,0 +1,165 @@
+                    # must be deleted before the bucket itself can be deleted.
+                    bucket.delete()
+
+    def delete_dynamodb_tables(self):
+        """Purge/Delete DynamoDB tables.
+
+        If tag was specified during object creating, then delete all tables tagged with this tag.
+        Else delete all _DEPLOYMENT_PREFIX_ prefixed tables.
+        """
+        client = boto3.client('dynamodb',
+                              aws_access_key_id=_AWS_ACCESS_KEY_ID,
+                              aws_secret_access_key=_AWS_SECRET_ACCESS_KEY,
+                              region_name=_AWS_DEFAULT_REGION)
+        resource = boto3.resource('dynamodb',
+                                  aws_access_key_id=_AWS_ACCESS_KEY_ID,
+                                  aws_secret_access_key=_AWS_SECRET_ACCESS_KEY,
+                                  region_name=_AWS_DEFAULT_REGION)
+
+        if self.tag:
+            print("About to delete DynamoDB tables tagged with %r." % ','.join(self.tag))
+        else:
+            print("About to delete %r prefixed DynamoDB tables." % _DEPLOYMENT_PREFIX)
+
+        for table in resource.tables.all():
+            match = False
+            if self.tag:
+                # table is either tagged with tag argument
+                match = {'Key': self.tag[0], 'Value': self.tag[1]} in\
+                        client.list_tags_of_resource(ResourceArn=table.table_arn).get('Tags', [])
+            # or table name is _DEPLOYMENT_PREFIX prefixed
+            elif table.name.startswith(_DEPLOYMENT_PREFIX + '_'):
+                match = True
+
+            if match:
+                # There's no 'purge' operation, so just delete the table in any case.
+                print("Deleting", table)
+                table.delete()
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-f", "--force",
+                        help="Allow to purge/delete resources tagged/prefixed with prod or stage")
+    parser.add_argument("-t", "--tag",
+                        help="Delete resources tagged with this 'key,value' tag")
+    args = parser.parse_args()
+
+    if not all(_AWS_ACCESS_KEY_ID and _AWS_SECRET_ACCESS_KEY and _AWS_DEFAULT_REGION and
+               _DEPLOYMENT_PREFIX):
+        print("Not all environment variables are properly defined.")
+        exit(1)
+
+    if args.tag:
+        try:
+            k, v = args.tag.split(',')
+            args.tag = (k, v)
+        except ValueError:
+            print("Tag needs to be 'key,value'")
+            exit(1)
+        print("About to delete resources tagged with %r" % ','.join(args.tag))
+    else:
+        print("About to purge %r prefixed resources." % _DEPLOYMENT_PREFIX)
+
+    if (args.tag and args.tag[1].lower() in {'prod', 'stage'}
+            or _DEPLOYMENT_PREFIX.lower() in {'prod', 'stage'}) and not args.force:
+        print("If you really want to destroy prod/stage use -f/--force.")
```

Comment:
🥇

Reply:
"If you really want to destroy {environment} use -f/--force".format(environment=args.tag[1] ?